react-app/src/components/Extractor.js

```diff
@@ -0,0 +1,312 @@
+import React, { useState } from 'react';
+import { Accordion, Button, Dropdown, Form } from 'react-bootstrap';
+import { Trash2 } from 'react-feather';
+import FilePicker from './FilePicker';
+
+function Extractor(props) {
+  const [extractorLabel, setExtractorLabel] = useState(props.formData.label ? props.formData.label : '');
+
+  // Variables for constructor arg management
+  function getArgValues() {
+    return [
+      {
+        filePath: props.formData.constructorArgs.filePath
+          ? props.formData.constructorArgs.filePath
+          : 'No File Selected',
+        label: 'CSV File Path',
+        type: 'file',
+        included: true,
+        key: 'filePath',
+        validExtractors: [],
+      },
+      {
+        url: props.formData.constructorArgs.url ? props.formData.constructorArgs.url : '',
+        label: 'CSV File URL',
+        type: 'url',
+        included: false,
+        key: 'url',
+        validExtractors: [],
+      },
+      {
+        clinicalSiteID: props.formData.constructorArgs.clinicalSiteID
+          ? props.formData.constructorArgs.clinicalSiteID
+          : '',
+        label: 'Clinical Site ID',
+        type: 'string',
+        included: false,
+        key: 'clinicalSiteID',
+        validExtractors: ['CSVClinicalTrialInformationExtractor'],
+      },
+      {
+        clinicalSiteSystem: props.formData.constructorArgs.clinicalSiteSystem
+          ? props.formData.constructorArgs.clinicalSiteSystem
+          : '',
+        label: 'Clinical Site System',
+        type: 'string',
+        included: false,
+        key: 'clinicalSiteSystem',
+        validExtractors: ['CSVClinicalTrialInformationExtractor'],
+      },
+      {
+        cancerType: props.formData.constructorArgs.cancerType ? props.formData.constructorArgs.cancerType : '',
+        label: 'Cancer Type',
+        type: 'string',
+        included: false,
+        key: 'cancerType',
+        validExtractors: ['CSVCancerDiseaseStatusExtractor'],
+      },
+      {
+        mask: props.formData.constructorArgs.mask ? props.formData.constructorArgs.mask : '',
+        label: 'Masked Fields',
+        type: 'dropdown',
+        included: false,
+        key: 'mask',
+        options: ['gender', 'mrn', 'name', 'address', 'birthDate', 'language', 'ethnicity', 'birthsex', 'race'],
+        validExtractors: ['CSVPatientExtractor'],
+      },
+    ];
+  }
+  const [args, setArgs] = useState(getArgValues());
+  const [argsJSX, setArgsJSX] = useState(
+    getArgValues()
+      .filter((arg) => arg.included === true)
+      .map((arg) => <div key={arg.key} />),
+  );
```

Comment:
I'm actually now thinking we don't need to manage this state variable at all. I think since `argsJSX` uses args to render the various inputs, we can actually just call `getArgsJSX` from within the render function. If we do that, then I don't think we need to save and keep track of the resulting value because we can just directly render it, and re-render it, when the data updates.

Reply:
I have removed the argsJSX variable.